contractcourt/htlc_timeout_resolver.go

```diff
@@ -483,13 +484,56 @@ func (h *htlcTimeoutResolver) sweepSecondLevelTx() error {
 			h.broadcastHeight,
 		))
 	}
+
+	nLocktime, required := inp.RequiredLockTime()
+	if !required {
+		log.Errorf("%T(%x): second level tx not timelocked", h,
+			h.htlc.RHash[:])
+	}
+
+	// Calculate the budget.
+	//
+	// TODO(yy): the budget is twice the output's value, which is needed as
+	// we don't force sweep the output now. To prevent cascading force
+	// closes, we use all its output value plus a wallet input as the
+	// budget. This is a temporary solution until we can optionally cancel
+	// the incoming HTLC, more details in,
+	// - https://github.com/lightningnetwork/lnd/issues/7969
+	budget := calculateBudget(
+		btcutil.Amount(inp.SignDesc().Output.Value), 2, 0,
+	)
+
+	// For an outgoing HTLC, it must be swept before the RefundTimeout of
+	// its incoming HTLC is reached.
+	//
+	// TODO(yy): find out the corresponding incoming HTLC's refund timeout.
+	// Here we just use `chainreg.DefaultBitcoinTimeLockDelta` as a
+	// starting point, and fix it in 0.18.1.
+	//
+	// TODO(yy): we may end up mixing inputs with different time locks.
+	// Suppose we have two outgoing HTLCs,
+	// - HTLC1: nLocktime is 800000, CLTV delta is 80.
+	// - HTLC2: nLocktime is 800001, CLTV delta is 79.
+	// This means they would both have an incoming HTLC that expires at
+	// 800080, hence they share the same deadline but different locktimes.
+	// However, with current design, when we are at block 800000, HTLC1 is
+	// offered to the sweeper. When block 800001 is reached, HTLC1's
+	// sweeping process is already started, while HTLC2 is being offered to
+	// the sweeper, so they won't be mixed. This can become an issue tho,
+	// if we decide to sweep per X blocks. Or the contractcourt sees the
+	// block first while the sweeper is only aware of the last block. To
+	// properly fix it, we need `blockbeat` to make sure subsystems are in
+	// sync.
+	deadline := nLocktime + 80
```

Comment:
Comment above mentions that `80` is the stop gap here to potentially address laterr on. We can likely just use the constant here also instead of `80`. 

In terms of obtaining the right value there isn't presently a good way to get the expiry of the incoming HTLC (as the scope here is a single channel). To do that, we'd need to add a new way to indirectly query the switch's circuit map, which actually doesn't directly contain the timeout values (not needed for forwarding), so once the incoming `CircuitKey` is known, we can use that to locate the link that has the HTLCs active on it. For the case when both the incoming _and_ outgoing channel are already in the chain for w/e reason, then we can query `ChainArbitrator`.

Reply:
Actually since the final HTLC state is replicated in the `ChannelArbitrator`, we can query that directly when we go to make the resolvers.